test/unit/mdc-list/foundation.test.js

```diff
@@ -383,7 +419,230 @@ test('#handleKeydown does not find ancestor with mdc-list-item so returns early'
   td.when(mockAdapter.getFocusedElementIndex()).thenReturn(-1);
   td.when(mockAdapter.getListItemIndex(td.matchers.anything())).thenReturn(-1);
   td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  td.when(mockAdapter.isListItem(td.matchers.anything())).thenReturn(false);
   foundation.handleKeydown(event);
 
   td.verify(preventDefault(), {times: 0});
 });
+
+test('#handleKeydown space key causes preventDefault to be called on the event', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const preventDefault = td.func('preventDefault');
+  const target = {classList: ['mdc-list-item']};
+  const event = {key: 'Space', target, preventDefault};
+
+  td.when(mockAdapter.getFocusedElementIndex()).thenReturn(0);
+  td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  foundation.setSingleSelection(true);
+  foundation.handleKeydown(event);
+
+  td.verify(preventDefault(), {times: 1});
+});
+
+test('#handleKeydown enter key causes preventDefault to be called on the event', () => {
```

Comment:
I think you have the titles of this and the next tests reversed with regards to `if singleSelection=true`